File: solutions/free-pascal/two-fer/1/TwoFer.pas
unit TwoFer;

{$mode ObjFPC}{$H+}

interface

function TwoFer(const name : string) : string;

implementation

function TwoFer(const name : string) : string;
begin
  if length(name) = 0 then
  begin
    result := 'One for ' + 'you' + ', one for me.';
  end
  else
  begin
    result := 'One for ' + name + ', one for me.';
  end;
end;

end.

File: solutions/objective-c/two-fer/1/TwoFer.m

#import <Foundation/Foundation.h>
#import "TwoFer.h"
@implementation TwoFer
+ (nonnull NSString *)twoFerWithName:(nullable NSString *)name {
    if (name != nil) {
        return [NSString stringWithFormat:@"One for %@, one for me.", name];
    } else {
        return @"One for you, one for me.";
    }
}
@end